### PR TITLE
Initialise application

### PR DESCRIPTION
**Pull request by wnowicki**
## Features
- Initialise Application

[Delivers#98740548]

### CODE REVIEW

File: app/Http/Controllers/InitialisationController.php [hunk 1]
```diff
@@ -0,0 +1,168 @@
...
+    }
+
+    /**
+     * @author WN
+     * @param int $locationId
+     * @return \Illuminate\View\View
+     */
+    public function prepare($locationId)
+    {
+        $this->fetchLocation($locationId);
```
Comment (sped2773) [inline app/Http/Controllers/InitialisationController.php]: Not sure why this is needed and it doesn't assign to anything

File: app/Http/Controllers/InitialisationController.php [hunk 2]
```diff
@@ -0,0 +1,168 @@
...
+        }
+    }
+
+    /**
+     * @author WN
+     * @param int $locationId
+     * @param Request $request
+     * @return \Illuminate\Http\JsonResponse
+     */
+    public function choseProduct($locationId, Request $request)
```
Comment (sped2773) [inline app/Http/Controllers/InitialisationController.php]: chooseProduct

File: app/Http/Controllers/InitialisationController.php (hunk 1 above)
Comment (wnowicki) [inline app/Http/Controllers/InitialisationController.php]: It's a standard helper. Main purpose is to fetch `Location` object. But is also checking if `user` has access to current `Merchant`

File: resources/views/initialise/main.blade.php [hunk 3]
```diff
@@ -0,0 +1,137 @@
...
+                <label>Price</label>
+
+                <div class="input-group">
+                    <div class="input-group-addon">£</div>
+                    {!! Form::text('amount', isset($amount)?number_format($amount/100,2):null, ['class' => 'form-control', 'maxlength' => 10]) !!}
+                </div>
+
+            </div>
+
+            <button type="submit" class="btn btn-primary">Show Finance Ooptions</button>
```
Comment (sped2773) [inline resources/views/initialise/main.blade.php]: Options

File: resources/views/initialise/main.blade.php [hunk 4]
```diff
@@ -0,0 +1,137 @@
...
+                                                <th>Payment Starts</th>
+                                                <td>{{ $product['credit_info']['payment_start_nice'] }}</td>
+                                            </tr>
+                                            <tr>
+                                                <th>Interest Rate (Fixed)</th>
+                                                <td>{{ number_format($product['credit_info']['offered_rate'], 2) }}%</td>
+                                            </tr>
+                                            <tr>
+                                                <th>Deposit</th>
+                                                <td>£{{ number_format($product['credit_info']['deposit_amount']/100, 2) }}</td>
```
Comment (sped2773) [inline resources/views/initialise/main.blade.php]: &pound;

Comment (wnowicki) [inline resources/views/initialise/main.blade.php]: not sure what you mean. need to discuss it in the morning.

Comment (wnowicki) [inline resources/views/initialise/main.blade.php]: done

File: resources/views/initialise/main.blade.php (hunk 3 above)
Comment (wnowicki) [inline resources/views/initialise/main.blade.php]: done

File: app/Http/Controllers/InitialisationController.php (hunk 2 above)
Comment (wnowicki) [inline app/Http/Controllers/InitialisationController.php]: done

**Comment by sped2773** [comment]
Need to resolve how we authenticate permissions for URLs but can be done as part of wider refactor.
 story.

**Pull request merged**